# Bioinfo_py_Scripts
## Python Scripts for Bioinformatics
### gc_percent.py
> This script take FASTA file as input and return the GC content
### extract_accession_no.py
> This script take MULTI-FASTA file as input and write all the Accession Number(s) in a new file (accession_no.txt)
### extract_fasta_headers.py
> This script take (Multi)Fasta file as input and write the Sequence Header(s) in a new file (fasta_headers.txt)
### k-mer_constructor.py
> This script take Fasta file as input and produce all the K-mers of length specified by the user
### random_seq_generator.py
> This script take sequence-type and sequence-length as input and produce random-nucleotide-sequence in FASTA format
### extract_fasta_records.py
> This script extracts Fasta-records from Multi-Fasta file whose Accession-No(s) are in Accession-Ids file
### fasta_record_finder.py
> This script extract Fasta-record from Multi-Fasta file whose Accession-No is inputted by the user and write the record in a new file (NC_XXXXXX.fasta)
### fasta_concatenator.py
> This script merge all the files with (.fasta) extension and create a new file (multi_fasta)
### multi_fasta_deconcatenator.py
> This script split multi fasta file into individual fasta file(s)
